File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        res = 0
        i = len(grid) - 1
        j = 0
        while i >= 0 and j < len(grid[0]):
            while j < len(grid[0]) and grid[i][j] >= 0:
                j += 1
            res += len(grid[0]) - j
            i -= 1
        return res

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int res = 0;
        int i = grid.size() - 1;
        int j = 0;
        while(i >= 0 && j < grid[0].size())
        {
            while(j < grid[0].size() && grid[i][j] >= 0)
            {
                j++;
            }
            res += grid[0].size() - j;
            i--;
        }
        return res;
    }
};